usb/usb.py

```diff
@@ -0,0 +1,245 @@
+    lines = lines.split("\n")
+    lines = filter(None, lines)
+    return lines
+
+def getKernelName(path):
+    return check_output(['lsblk', '-ndso', 'KNAME', path],
+            universal_newlines=True).rstrip("\n")
+
+def getDeviceType(path):
+    return check_output(['lsblk', '-no', 'TYPE', path],
+            universal_newlines=True).strip() 
+
+def getUnlockedCryptOutput(path):
+    return path
+
+def getFSType(path):
+    global attributeMaps
+    return attributeMaps[path].get("ID_FS_TYPE")
+
+def isLUKSPartition(path):
+    return getFSType(path) == "crypto_LUKS"
+
+def isSwapPartition(path):
+    return getFSType(path) == "swap"
+
+def getFSLabel(path):
+    global attributeMaps
+    label = attributeMaps[path].get("ID_FS_LABEL_ENC", "")
+    if label:
+        label = label.encode().decode("unicode-escape")
+        if type(TRUNCATE_FS_LABELS) == int:
+            if TRUNCATE_FS_LABELS >= 0:
+                label = label[:TRUNCATE_FS_LABELS]
+            elif TRUNCATE_FS_LABELS < 0:
+                label = label[TRUNCATE_FS_LABELS:]
+    return label
+
+def getFSOptions(path):
+    lines = check_output(['findmnt', '-no', 'FS-OPTIONS', path],
+            universal_newlines=True).strip()
+    lines = lines.split(",")
+    return lines
+
+def isReadOnly(path):
+    return "ro" in getFSOptions(path)
+
+def isExtendedPartitionMarker(path):
+    global attributeMaps
+    MARKERS = ["0xf", "0x5"]
+    return attributeMaps[path].get("ID_PART_ENTRY_TYPE") in MARKERS
+
+def getMountPoint(path):
+    return check_output(['lsblk', '-ndo', 'MOUNTPOINT', path],
+            universal_newlines=True).rstrip("\n")
+
+def getSpaceAvailable(path):
+    lines = check_output(['df', '-h', '--output=avail', path],
+            universal_newlines=True)
+    lines = lines.split("\n")
+    if len(lines) != 3:
+        return ""
+    else:
+        return lines[1].strip()
+
+def getLockedCryptOutput(path):
+    form = "<span color='{}'>[{} {}]</span>"
+    kname = getKernelName(path)
+    output = form.format(LOCKED_COLOR, FA_LOCK, kname)
+    return output
+
+def getParentKernelName(path):
+    lines = check_output(['lsblk', '-nso', 'KNAME', path],
+            universal_newlines=True)
+    lines = lines.split("\n")
+    if len(lines) > 2:
+        return lines[1].rstrip("\n")
+    else:
+        return ""
+
+def getUnlockedCryptOutput(path):
+    mountPoint = getMountPoint(path)
+    if mountPoint:
+        color = MOUNTED_COLOR
+        if isReadOnly(path):
+            spaceAvail = "ro"
+        else:
+            spaceAvail = getSpaceAvailable(path)
+        mountPoint = "<i>{}</i>:".format(pangoEscape(mountPoint))
+    else:
+        color = UNLOCKED_NOT_MOUNTED_COLOR
+        spaceAvail = ""
+    kernelName = getKernelName(path)
+    parentKernelName = getParentKernelName(path)
+    
+    block = "<span color='{}'>[{} {}:{}]</span>"
+    block = block.format(color, FA_UNLOCK, parentKernelName, kernelName)
```

Comment:
As mentioned above `getSpaceAvailable` uses `df -h --output=avail` which I don't believe needs to be escaped. However, I could see how escaping just in case might be wanted. This is where I would do it. Also as far as I know a kernel name will never have bad characters  in it either. You know as I try to explain these things I feel like guarding against bad characters should be done anyway.